### PR TITLE
run the generate scripts only when the source has changed.

### PR DESCRIPTION
**Pull request by krasi-georgiev**
@themandalore  This should better answer your questions about PHONY targets.

Signed-off-by: Krasi Georgiev <8903888+krasi-georgiev@users.noreply.github.com>

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -52,20 +52,18 @@ deps: ## Ensures fresh go.mod and go.sum.
 
 .PHONY: generate
 generate: ## Generate all dynamic files.
-generate: generate-pow generate-sol
+generate: pkg/pow/kernelSource.go pkg/contracts/%/%
```
Comment (krasi-georgiev) [inline Makefile:55]: this means run a target called `pkg/pow/kernelSource.go` , `pkg/contracts/%/%` 

`%/%` is a wild char to match `pkg/contracts/tellor/tellor.go` and `pkg/contracts/getter/getter.go`

File: Makefile [hunk 2]
```diff
@@ -52,20 +52,18 @@ deps: ## Ensures fresh go.mod and go.sum.
...
 
-.PHONY: generate-pow
-generate-pow:
-	@go run ./scripts/opencl
+pkg/pow/kernelSource.go: scripts/opencl/sources/*
```
Comment (krasi-georgiev) [inline Makefile:57]: this means: run this target only if `pkg/pow/kernelSource.go` is missing or any of the files in `scripts/opencl/sources/` has changed.

File: Makefile [hunk 3]
```diff
@@ -52,20 +52,18 @@ deps: ## Ensures fresh go.mod and go.sum.
...
+	go run ./scripts/opencl
 
-.PHONY: generate-sol
-generate-sol: $(SOLCCHECK) $(ABIGEN)
-	@rm -Rf pkg/contracts/
+pkg/contracts/tellor/tellor.go pkg/contracts/getter/getter.go: contracts/* $(SOLCCHECK) $(ABIGEN)
```
Comment (krasi-georgiev) [inline Makefile:60]: this means: run this target only if `pkg/contracts/tellor/tellor.go` or  `pkg/contracts/getter/getter.go` are missing or if any of the files in `contracts/*` has changed.

lets ignore the explanation for `$(SOLCCHECK) $(ABIGEN)` for now :+1:

File: Makefile [hunk 4]
```diff
@@ -52,20 +52,18 @@ deps: ## Ensures fresh go.mod and go.sum.
 
 .PHONY: generate
 generate: ## Generate all dynamic files.
-generate: generate-pow generate-sol
+generate: pkg/pow/kernelSource.go pkg/contracts/tellor/tellor.go pkg/contracts/getter/getter.go
```
Comment (krasi-georgiev) [inline Makefile:55]: this means: run targets 
 - pkg/pow/kernelSource.go 
 - pkg/contracts/tellor/tellor.go 
 - pkg/contracts/getter/getter.go

**Review by themandalore — APPROVED** [review]

**Pull request merged**